src/docker/health/http_checker.go

```diff
@@ -25,8 +29,28 @@ func (checker *HTTPChecker) Check() Status {
 
 	client := checker.Client
 	if client == nil {
+		dialContext := func(ctx context.Context, network, addr string) (net.Conn, error) {
+			conn, err := net.Dial(network, addr)
+			if err != nil {
+				return nil, err
+			}
+
+			header := proxyproto.Header{
+				Command: proxyproto.LOCAL,
+				Version: 2,
+			}
+			_, err = header.WriteTo(conn)
+			if err != nil {
+				return nil, err
+			}
+			return conn, nil
+		}
+		if !checker.ProxySupport {
+			dialContext = nil
+		}
```

Comment:
All of this header-injection logic should be initialized once in a HTTP client that is injected into this `HTTPChecker` struct in [src/hcmd/healthcheck/main.go](https://github.com/icecave/honeycomb/blob/master/src/cmd/healthcheck/main.go). The checker itself needn't be concerned with whether or not proxy protocol is being used.

It'd be good to pull the header-writing code out into a named function instead of a closure, in that case.

Reply:
Moved the client creation into src/cmd/healthcheck/main.go, also removed the PROXY from the test cases for the healthcheck.